embedded/embedded-api/src/main/java/org/mule/runtime/module/embedded/internal/Repository.java

```diff
@@ -0,0 +1,209 @@
+
+  private DefaultRepositorySystemSession session;
+  private RepositorySystem system;
+
+  public Repository() {
+    createRepositorySystem();
+  }
+
+  private void createRepositorySystem() {
+    session = newDefaultRepositorySystemSession();
+    session.setOffline(true);
+    session.setIgnoreArtifactDescriptorRepositories(true);
+
+    File mavenLocalRepositoryLocation = new File(MAVEN_REPOSITORY_FOLDER);
+    system = newRepositorySystem(mavenLocalRepositoryLocation, session);
+  }
+
+  private static DefaultRepositorySystemSession newDefaultRepositorySystemSession() {
+    final DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
+    session.setUpdatePolicy(UPDATE_POLICY_NEVER);
+    session.setChecksumPolicy(CHECKSUM_POLICY_IGNORE);
+
+    DependencySelector dependencySelector =
+        new AndDependencySelector(new ScopeDependencySelector(RUNTIME, TEST),
+                                  new OptionalDependencySelector(),
+                                  new ExclusionDependencySelector());
+    session.setDependencySelector(dependencySelector);
+
+    return session;
+  }
+
+  public PreorderNodeListGenerator assemblyDependenciesForArtifact(Artifact artifact, Predicate<Dependency> filter) {
+    final CollectRequest collectRequest = new CollectRequest();
+    try {
+      final ArtifactDescriptorResult artifactDescriptorResult =
+          system.readArtifactDescriptor(session,
+                                        new ArtifactDescriptorRequest(artifact, null, null));
+      collectRequest.setDependencies(artifactDescriptorResult.getDependencies()
+          .stream()
+          .filter(dependency -> filter.test(dependency))
+          .collect(toList()));
+      collectRequest.setManagedDependencies(artifactDescriptorResult.getManagedDependencies());
+
+      final CollectResult collectResult = system.collectDependencies(session, collectRequest);
+
+      final DependencyRequest dependencyRequest = new DependencyRequest();
+
+      Collection<String> excluded = new ArrayList<>();
+      Collections.addAll(excluded, PROVIDED, SYSTEM, RUNTIME, TEST);
+      Collection<String> included = new ArrayList<>();
+      included.add(COMPILE);
+      dependencyRequest.setFilter(new ScopeDependencyFilter(included, excluded));
+
+      dependencyRequest.setRoot(collectResult.getRoot());
+      dependencyRequest.setCollectRequest(collectRequest);
+      final DependencyResult dependencyResult =
+          system.resolveDependencies(session, dependencyRequest);
+      final PreorderNodeListGenerator nlg = new PreorderNodeListGenerator();
+      dependencyResult.getRoot().accept(nlg);
+      return nlg;
+    } catch (DependencyResolutionException e) {
+      DependencyNode node = e.getResult().getRoot();
+      logUnresolvedArtifacts(node, e);
+      throw new RuntimeException(
+                                 String.format("There was an issue solving the dependencies for the container [%s]",
```

Comment:
I changed it to "problem". I won't to mention that there's a problem more than saying that there was an exception.